src/librustc/mir/mod.rs

```diff
@@ -2248,7 +2248,7 @@ pub enum CastKind {
     ReifyFnPointer,
 
     /// Converts non capturing closure to fn()
-    ClosureFnPointer,
```

Comment:
Can you extend doc-comment here as well? Same aspect about what the body of the closure can and cannot do would be good info.